keras_hub/src/models/yolo_v8/ciou_loss.py

```diff
@@ -0,0 +1,92 @@
+import keras
+from keras import ops
+
+from keras_hub.src.bounding_box.iou import compute_ciou
+
+
+class CIoULoss(keras.losses.Loss):
+    """Implements the Complete IoU (CIoU) Loss
+
+    CIoU loss is an extension of GIoU loss, which further improves the IoU
+    optimization for object detection. CIoU loss not only penalizes the
+    bounding box coordinates but also considers the aspect ratio and center
+    distance of the boxes. The length of the last dimension should be 4 to
+    represent the bounding boxes.
+
+    Args:
+        bounding_box_format: a case-insensitive string (for example, "xyxy").
+            Each bounding box is defined by these 4 values. For detailed
+            information on the supported formats, see the [Keras bounding box
+            documentation](https://github.com/keras-team/keras/blob/master/
+            keras/src/layers/preprocessing/image_preprocessing/
+            bounding_boxes/formats.py).
+        epsilon: A small value added to avoid division by zero and stabilize
+            calculations.
+
+    References:
+        - [CIoU paper](https://arxiv.org/pdf/2005.03572.pdf)
+
+    Example:
+    ```python
+    y_true = np.random.uniform(
+        size=(5, 10, 4),
+        low=0,
+        high=10)
+    y_pred = np.random.uniform(
+        size=(5, 10, 4),
+        low=0,
+        high=10)
+    loss = keras_hub.src.models.yolo_v8.ciou_loss.CIoULoss("xyxy")
+    loss(y_true, y_pred).numpy()
+    ```
+
+    Usage with the `compile()` API:
+    ```python
+    model.compile(optimizer='adam', loss=CIoULoss())
+    ```
+    """
+
+    def __init__(self, bounding_box_format, epsilon=1e-7, **kwargs):
```

Comment:
should this default to `None` and use `keras.config.epsilon()`?

Reply:
Yes, I think that would be better. I have now changed it to default to the core Keras epsilon.